eclipse-theia/theia
# Don't set `viewItem` context on selection

#### What it does


Depends on #11741.

Fixes #11458 by using a transient context for tree view context menus rather than a permanent setting when the selection changes in the tree view.

#### How to test


1. Install [this plugin](https://github.com/eclipse-theia/theia/files/9729277/custom-view-samples-0.0.1.zip). ([Source](https://github.com/colin-grant-work/vscode-extension-samples/tree/demo/menu-items/tree-view-sample)).
2. Open the `Test View` in the Explorer view container.
3. Observe that the two parent items show a sub-type hierarchy icon, and clicking it shows a message about clicking a one-letter item.
4. Observe that the context menus for those items have the same item, and activating it shows the same message.
5. Observe that child items with the letter 'b' in their labels have a super-type hierarchy icon, and clicking it shows a message about clicking on an item with the letter 'b'.
6. Observe that the same item appears in the context menus for those items, and activating it shows the same message.
7. Observe that the child items do _not_ have the parent item's icon, even when the parent item is selected.


#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
